frontend/src/components/dashboard/create/email/EmailRecipients.tsx

```diff
@@ -23,39 +24,63 @@ const EmailRecipients = ({
   csvFilename: initialCsvFilename,
   numRecipients: initialNumRecipients,
   params,
+  isProcessing: initialIsProcessing,
   onNext,
 }: {
   csvFilename: string
   numRecipients: number
   params: Array<string>
-  onNext: (changes: any, next?: boolean) => void
+  isProcessing: boolean
+  onNext: (changes: Partial<EmailCampaign>, next?: boolean) => void
 }) => {
   const [errorMessage, setErrorMessage] = useState(null)
-  const [csvFilename, setUploadedCsvFilename] = useState(initialCsvFilename)
-  const [numRecipients, setNumRecipients] = useState(initialNumRecipients)
+  const [isCsvProcessing, setIsCsvProcessing] = useState(initialIsProcessing)
   const [isUploading, setIsUploading] = useState(false)
-  const [preview, setPreview] = useState(
-    {} as { body: string; subject: string; reply_to: string | null }
-  )
-
+  const [csvInfo, setCsvInfo] = useState<
+    Omit<CsvStatusResponse, 'isCsvProcessing' | 'preview'>
+  >({
+    numRecipients: initialNumRecipients,
+    csvFilename: initialCsvFilename,
+  })
+  const [preview, setPreview] = useState({} as EmailPreview)
   const { id: campaignId } = useParams()
 
-  async function loadPreview(campaignId: string) {
-    try {
-      const msgPreview = await getPreviewMessage(+campaignId)
-      if (msgPreview) {
-        setPreview(msgPreview)
+  const { csvFilename, numRecipients = 0 } = csvInfo
+
+  // Poll csv status
+  useEffect(() => {
+    if (!campaignId) return
+
+    let timeoutId: NodeJS.Timeout
+    const pollStatus = async () => {
+      try {
+        const { isCsvProcessing, preview, ...newCsvInfo } = await getCsvStatus(
+          +campaignId
+        )
+        setIsCsvProcessing(isCsvProcessing)
+        setCsvInfo(newCsvInfo)
+        preview && setPreview(preview as EmailPreview)
```

Comment:
how did this pass linting... this means `if (preview) setPreview(preview)` right

Reply:
I guess it's the react way
https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator